Cannot build attoparsec-iso8601 on master

When attempting to build the `attoparsec-iso8601` library on commit 6bd27358bf0e990fdab339158f925452e8900c3d, I get the following build error:

```
$ /opt/cabal/3.0/bin/cabal build attoparsec-iso8601
Build profile: -w ghc-8.8.3 -O1
In order, the following will be built (use -v for more details):
 - attoparsec-iso8601-1.0.1.0 (lib) (first run)
Preprocessing library for attoparsec-iso8601-1.0.1.0..
Building library for attoparsec-iso8601-1.0.1.0..
[1 of 2] Compiling Data.Attoparsec.Time.Internal ( Data/Attoparsec/Time/Internal.hs, /home/rgscott/Documents/Hacking/Haskell/aeson/dist-newstyle/build/x86_64-linux/ghc-8.8.3/attoparsec-iso8601-1.0.1.0/build/Data/Attoparsec/Time/Internal.o )

Data/Attoparsec/Time/Internal.hs:24:1: error:
    Could not load module ‘Data.Time.Clock.Compat’
    It is a member of the hidden package ‘time-compat-1.9.3’.
    Perhaps you need to add ‘time-compat’ to the build-depends in your .cabal file.
    It is a member of the hidden package ‘time-compat-1.9.3’.
    Perhaps you need to add ‘time-compat’ to the build-depends in your .cabal file.
    It is a member of the hidden package ‘time-compat-1.9.3’.
    Perhaps you need to add ‘time-compat’ to the build-depends in your .cabal file.
    It is a member of the hidden package ‘time-compat-1.9.2.2’.
    Perhaps you need to add ‘time-compat’ to the build-depends in your .cabal file.
    It is a member of the hidden package ‘time-compat-1.9.2.2’.
    Perhaps you need to add ‘time-compat’ to the build-depends in your .cabal file.
    It is a member of the hidden package ‘time-compat-1.9.2.2’.
    Perhaps you need to add ‘time-compat’ to the build-depends in your .cabal file.
    Use -v (or `:set -v` in ghci) to see a list of the files searched for.
   |
24 | import Data.Time.Clock.Compat
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Is there a missing dependency on `time-compat` somewhere?

--- a/.travis.yml
+++ b/.travis.yml
@@ -25,21 +25,34 @@ before_cache:
 matrix:
   include:
 
+    # Do the special build steps and handle the newest and oldest
+    # GHCs first, since these are the most likely to fail.
+
+    # HLint
     - env: BUILD=hlint CABALVER=2.2 GHCVER=8.4.4
       compiler: ": #GHC 8.4.4"
       addons: {apt: {packages: [cabal-install-2.2,ghc-8.4.4], sources: [hvr-ghc]}}
 
+    # Latest GHC
+    - env: BUILD=cabal2 CABALVER=3.0 GHCVER=8.10.1
+      compiler: ": #GHC 8.10.1"
+      addons: {apt: {packages: [cabal-install-3.0,ghc-8.10.1], sources: [hvr-ghc]}}
+
+    # Oldest GHC
+    - env: BUILD=cabal CABALVER=1.18 GHCVER=7.8.4
+      compiler: ": #GHC 7.8.4"
+      addons: {apt: {packages: [cabal-install-1.18,ghc-7.8.4], sources: [hvr-ghc]}}
+
+    # Benchmarks
     - env: BUILD=stack CABALVER=2.4 GHCVER=8.6.5 STACK_YAML=stack-bench.yaml
       compiler: ": #GHC 8.6.5"
       addons: {apt: {packages: [cabal-install-2.4,ghc-8.6.5], sources: [hvr-ghc]}}
 
+    # C unescape implementation
     - env: BUILD=stack CABALVER=2.2 GHCVER=8.6.5 STACK_YAML=stack-ffi-unescape.yaml
       compiler: ": #GHC 8.6.5"
       addons: {apt: {packages: [cabal-install-2.2,ghc-8.6.5], sources: [hvr-ghc]}}
 
-    - env: BUILD=cabal2 CABALVER=3.0 GHCVER=8.10.1
-      compiler: ": #GHC 8.10.1"
-      addons: {apt: {packages: [cabal-install-3.0,ghc-8.10.1], sources: [hvr-ghc]}}
 
     - env: BUILD=cabal2 CABALVER=3.0 GHCVER=8.8.3
       compiler: ": #GHC 8.8.3"
@@ -64,10 +77,6 @@ matrix:
     - env: BUILD=cabal CABALVER=1.22 GHCVER=7.10.3
       compiler: ": #GHC 7.10.3"
       addons: {apt: {packages: [cabal-install-1.22,ghc-7.10.3], sources: [hvr-ghc]}}
-
-    - env: BUILD=cabal CABALVER=1.18 GHCVER=7.8.4
-      compiler: ": #GHC 7.8.4"
-      addons: {apt: {packages: [cabal-install-1.18,ghc-7.8.4], sources: [hvr-ghc]}}
 
 before_install:
  - unset CC

--- a/.travis.yml
+++ b/.travis.yml
@@ -33,11 +33,6 @@ matrix:
       compiler: ": #GHC 8.4.4"
       addons: {apt: {packages: [cabal-install-2.2,ghc-8.4.4], sources: [hvr-ghc]}}
 
-    # attoparsec-iso8601
-    - env: BUILD=attoparsec-iso8601 CABALVER=3.0 GHCVER=8.10.1
-      compiler: ": #GHC 8.10.1"
-      addons: {apt: {packages: [cabal-install-3.0,ghc-8.10.1], sources: [hvr-ghc]}}
-
     # Latest GHC
     - env: BUILD=cabal2 CABALVER=3.0 GHCVER=8.10.1
       compiler: ": #GHC 8.10.1"
@@ -49,33 +44,33 @@ matrix:
       addons: {apt: {packages: [cabal-install-1.18,ghc-7.8.4], sources: [hvr-ghc]}}
 
     # Benchmarks
-    - env: BUILD=stack CABALVER=2.4 GHCVER=8.6.5 STACK_YAML=stack-bench.yaml
+    - env: BUILD=stack CABALVER=3.0 GHCVER=8.6.5 STACK_YAML=stack-bench.yaml
       compiler: ": #GHC 8.6.5"
-      addons: {apt: {packages: [cabal-install-2.4,ghc-8.6.5], sources: [hvr-ghc]}}
+      addons: {apt: {packages: [cabal-install-3.0,ghc-8.6.5], sources: [hvr-ghc]}}
 
     # C unescape implementation
-    - env: BUILD=stack CABALVER=2.2 GHCVER=8.6.5 STACK_YAML=stack-ffi-unescape.yaml
+    - env: BUILD=stack CABALVER=3.0 GHCVER=8.6.5 STACK_YAML=stack-ffi-unescape.yaml
       compiler: ": #GHC 8.6.5"
-      addons: {apt: {packages: [cabal-install-2.2,ghc-8.6.5], sources: [hvr-ghc]}}
+      addons: {apt: {packages: [cabal-install-3.0,ghc-8.6.5], sources: [hvr-ghc]}}
 
 
     - env: BUILD=cabal2 CABALVER=3.0 GHCVER=8.8.3
       compiler: ": #GHC 8.8.3"
       addons: {apt: {packages: [cabal-install-3.0,ghc-8.8.3], sources: [hvr-ghc]}}
 
-    - env: BUILD=cabal2 CABALVER=2.4 GHCVER=8.6.5
+    - env: BUILD=cabal2 CABALVER=3.0 GHCVER=8.6.5
       compiler: ": #GHC 8.6.5"
       addons: {apt: {packages: [cabal-install-3.0,ghc-8.6.5], sources: [hvr-ghc]}}
 
-    - env: BUILD=cabal2 CABALVER=2.2 GHCVER=8.4.4
+    - env: BUILD=cabal2 CABALVER=3.0 GHCVER=8.4.4
       compiler: ": #GHC 8.4.4"
       addons: {apt: {packages: [cabal-install-3.0,ghc-8.4.4], sources: [hvr-ghc]}}
 
-    - env: BUILD=cabal2 CABALVER=2.0 GHCVER=8.2.2
+    - env: BUILD=cabal2 CABALVER=3.0 GHCVER=8.2.2
       compiler: ": #GHC 8.2.2"
       addons: {apt: {packages: [cabal-install-3.0,ghc-8.2.2], sources: [hvr-ghc]}}
 
-    - env: BUILD=cabal2 CABALVER=2.0 GHCVER=8.0.2
+    - env: BUILD=cabal2 CABALVER=3.0 GHCVER=8.0.2
       compiler: ": #GHC 8.0.2"
       addons: {apt: {packages: [cabal-install-3.0,ghc-8.0.2], sources: [hvr-ghc]}}
 

--- a/.travis.yml
+++ b/.travis.yml
@@ -58,19 +58,19 @@ matrix:
       compiler: ": #GHC 8.8.3"
       addons: {apt: {packages: [cabal-install-3.0,ghc-8.8.3], sources: [hvr-ghc]}}
 
-    - env: BUILD=cabal CABALVER=2.4 GHCVER=8.6.5
+    - env: BUILD=cabal2 CABALVER=2.4 GHCVER=8.6.5
       compiler: ": #GHC 8.6.5"
       addons: {apt: {packages: [cabal-install-2.4,ghc-8.6.5], sources: [hvr-ghc]}}
 
-    - env: BUILD=cabal CABALVER=2.2 GHCVER=8.4.4
+    - env: BUILD=cabal2 CABALVER=2.2 GHCVER=8.4.4
       compiler: ": #GHC 8.4.4"
       addons: {apt: {packages: [cabal-install-2.2,ghc-8.4.4], sources: [hvr-ghc]}}
 
-    - env: BUILD=cabal CABALVER=2.0 GHCVER=8.2.2
+    - env: BUILD=cabal2 CABALVER=2.0 GHCVER=8.2.2
       compiler: ": #GHC 8.2.2"
       addons: {apt: {packages: [cabal-install-2.0,ghc-8.2.2], sources: [hvr-ghc]}}
 
-    - env: BUILD=cabal CABALVER=2.0 GHCVER=8.0.2
+    - env: BUILD=cabal2 CABALVER=2.0 GHCVER=8.0.2
       compiler: ": #GHC 8.0.2"
       addons: {apt: {packages: [cabal-install-2.0,ghc-8.0.2], sources: [hvr-ghc]}}
 

--- a/.travis.yml
+++ b/.travis.yml
@@ -33,6 +33,11 @@ matrix:
       compiler: ": #GHC 8.4.4"
       addons: {apt: {packages: [cabal-install-2.2,ghc-8.4.4], sources: [hvr-ghc]}}
 
+    # attoparsec-iso8601
+    - env: BUILD=attoparsec-iso8601 CABALVER=3.0 GHCVER=8.10.1
+      compiler: ": #GHC 8.10.1"
+      addons: {apt: {packages: [cabal-install-3.0,ghc-8.10.1], sources: [hvr-ghc]}}
+
     # Latest GHC
     - env: BUILD=cabal2 CABALVER=3.0 GHCVER=8.10.1
       compiler: ": #GHC 8.10.1"

--- a/.travis.yml
+++ b/.travis.yml
@@ -65,19 +65,19 @@ matrix:
 
     - env: BUILD=cabal2 CABALVER=2.4 GHCVER=8.6.5
       compiler: ": #GHC 8.6.5"
-      addons: {apt: {packages: [cabal-install-2.4,ghc-8.6.5], sources: [hvr-ghc]}}
+      addons: {apt: {packages: [cabal-install-3.0,ghc-8.6.5], sources: [hvr-ghc]}}
 
     - env: BUILD=cabal2 CABALVER=2.2 GHCVER=8.4.4
       compiler: ": #GHC 8.4.4"
-      addons: {apt: {packages: [cabal-install-2.2,ghc-8.4.4], sources: [hvr-ghc]}}
+      addons: {apt: {packages: [cabal-install-3.0,ghc-8.4.4], sources: [hvr-ghc]}}
 
     - env: BUILD=cabal2 CABALVER=2.0 GHCVER=8.2.2
       compiler: ": #GHC 8.2.2"
-      addons: {apt: {packages: [cabal-install-2.0,ghc-8.2.2], sources: [hvr-ghc]}}
+      addons: {apt: {packages: [cabal-install-3.0,ghc-8.2.2], sources: [hvr-ghc]}}
 
     - env: BUILD=cabal2 CABALVER=2.0 GHCVER=8.0.2
       compiler: ": #GHC 8.0.2"
-      addons: {apt: {packages: [cabal-install-2.0,ghc-8.0.2], sources: [hvr-ghc]}}
+      addons: {apt: {packages: [cabal-install-3.0,ghc-8.0.2], sources: [hvr-ghc]}}
 
     - env: BUILD=cabal CABALVER=1.22 GHCVER=7.10.3
       compiler: ": #GHC 7.10.3"

--- a/attoparsec-iso8601/attoparsec-iso8601.cabal
+++ b/attoparsec-iso8601/attoparsec-iso8601.cabal
@@ -35,11 +35,12 @@ library
     Data.Attoparsec.Time.Internal
     Data.Attoparsec.Time
   build-depends:
-    attoparsec >= 0.13.0.1,
+    attoparsec >= 0.13.0.1 && < 0.14.0.0,
     base >= 4.7 && < 5,
-    base-compat >= 0.9.1 && < 0.12,
-    text >= 1.1.1.0,
-    time >= 1.1.1.4
+    base-compat-batteries >= 0.10.0 && < 0.12,
+    time-compat >= 1.9.2.2 && < 1.10,
+    text >= 1.1.1.0 && < 1.3.0.0,
+    time >= 1.1.1.4 && < 1.11
 
   if flag(fast)
     ghc-options: -O0

--- a/travis/install.sh
+++ b/travis/install.sh
@@ -50,9 +50,14 @@ case $BUILD in
     fi
     ;;
   cabal2)
-	cabal v2-update -v
+	  cabal v2-update -v
     sed -i 's/^jobs:/-- jobs:/' ${HOME}/.cabal/config
 
-	cabal v2-build --only-dependencies --enable-tests --enable-benchmarks --dry
-	;;
+	  cabal v2-build --only-dependencies --enable-tests --enable-benchmarks --dry
+  	;;
+  attoparsec-iso8601)
+    cd attoparsec-iso8601
+    cabal v2-update -v
+    sed -i 's/^jobs:/-- jobs:/' ${HOME}/.cabal/config
+    cabal v2-build --only-dependencies --enable-tests --enable-benchmarks --dry
 esac

--- a/travis/script.sh
+++ b/travis/script.sh
@@ -20,12 +20,8 @@ case $BUILD in
       (cd dist && cabal install --force-reinstalls "$SRC_TGZ")
     ;;
   cabal2)
-    cabal v2-build --enable-tests --enable-benchmarks
-    cabal v2-test --enable-tests --enable-benchmarks
-    ;;
-  attoparsec-iso8601)
-    cabal v2-build --enable-tests --enable-benchmarks
-    cabal v2-test --enable-tests --enable-benchmarks
+    cabal v2-build all --enable-tests --enable-benchmarks
+    cabal v2-test all --enable-tests --enable-benchmarks
     ;;
   hlint)
     make lint

--- a/travis/script.sh
+++ b/travis/script.sh
@@ -19,7 +19,11 @@ case $BUILD in
     SRC_TGZ=$(cabal info . | awk '{print $2;exit}').tar.gz &&
       (cd dist && cabal install --force-reinstalls "$SRC_TGZ")
     ;;
-  cabal2) 
+  cabal2)
+    cabal v2-build --enable-tests --enable-benchmarks
+    cabal v2-test --enable-tests --enable-benchmarks
+    ;;
+  attoparsec-iso8601)
     cabal v2-build --enable-tests --enable-benchmarks
     cabal v2-test --enable-tests --enable-benchmarks
     ;;

